packages/code-surfer/src/step-parser.js

```diff
@@ -1,32 +1,14 @@
-const mapRange = range => {
-  if (!range) return {};
-  const newTokens = {};
-  const [start, end] = range;
-  for (let i = start; i <= end; i++) {
-    newTokens[i] = null;
-  }
-  return newTokens;
-};
-const mapLines = lines => {
-  const newTokens = {};
-  lines.forEach(line => (newTokens[line] = null));
-  return newTokens;
-};
-
-export const getTokensPerLine = ({
-  lines = [],
-  range,
-  ranges = [],
-  tokens
-}) => {
-  const newTokens = {};
-
-  Object.assign(newTokens, mapLines(lines));
-  Object.assign(newTokens, mapRange(range));
-  Object.assign(newTokens, ...ranges.map(mapRange));
-  Object.assign(newTokens, tokens);
-
-  return newTokens;
-};
-
-export default getTokensPerLine;
+const mapRange = (range, obj = {}) =>
+  !range || range[0] > range[1]
+    ? obj
+    : mapRange([range[0] + 1, range[1]], { ...obj, [range[0]]: null });
+
+const mapLines = lines =>
+  lines.reduce((obj, line) => ({ ...obj, [line]: null }), {});
+
+export default ({ lines = [], range, ranges = [], tokens }) => ({
+  ...mapLines(lines),
+  ...mapRange(range),
+  ...Object.assign({}, ...ranges.map(mapRange)),
```

Comment:
I would _really_ like to get rid of this `Object.assign()` as well, but don't know how 😅

Reply:
ES6ify the file !